cmd/ocitools/generate.go

```diff
@@ -96,8 +96,13 @@ var generateCommand = cli.Command{
 	},
 }
 
-func setupSpec(g generate.Generator, context *cli.Context) error {
-	spec := g.GetSpec()
+func setupSpec(gp *generate.Generator, context *cli.Context) error {
+	if context.GlobalBool("host-specific") {
+		gp.HostSpecific = true
+	}
+
+	g := *gp
```

Comment:
I don't think we need this.  It seemed to compile and run fine for me with the following change on top of 751e18b91:

    diff --git a/cmd/ocitools/generate.go b/cmd/ocitools/generate.go
    index ab1f6cc..a6423a0 100644
    --- a/cmd/ocitools/generate.go
    +++ b/cmd/ocitools/generate.go
    @@ -96,12 +96,11 @@ var generateCommand = cli.Command{
     	},
     }
     
    -func setupSpec(gp *generate.Generator, context *cli.Context) error {
    +func setupSpec(g *generate.Generator, context *cli.Context) error {
     	if context.GlobalBool("host-specific") {
    -		gp.HostSpecific = true
    +		g.HostSpecific = true
     	}
     
    -	g := *gp
     	spec := g.Spec()
     
     	if len(spec.Version) == 0 {
    @@ -374,7 +373,7 @@ func checkNs(nsMaps map[string]string, nsName string) bool {
     	return true
     }
     
    -func setupLinuxNamespaces(g generate.Generator, needsNewUser bool, nsMaps map[string]string) {
    +func setupLinuxNamespaces(g *generate.Generator, needsNewUser bool, nsMaps map[string]string) {
     	for _, nsName := range generate.Namespaces {
     		if !checkNs(nsMaps, nsName) && !(needsNewUser && nsName == "user") {
     			continue